File: app-python/entitlements_app/requests/add_feature_to_plan_http_request.py
"""Caller-visible HTTP body for command `AddFeatureToPlan`.

Request-context fields are supplied by authenticated transport state, not caller JSON.
"""

from pydantic import BaseModel


class AddFeatureToPlanHttpRequest(BaseModel):
    allowance: int
    effectiveAt: str
    enforcementMode: str
    entitlementId: str
    featureKey: str
    tenantId: str
    unit: str

    @property
    def effective_at(self) -> str:
        return self.effectiveAt

    @property
    def enforcement_mode(self) -> str:
        return self.enforcementMode

    @property
    def entitlement_id(self) -> str:
        return self.entitlementId

    @property
    def feature_key(self) -> str:
        return self.featureKey

    @property
    def tenant_id(self) -> str:
        return self.tenantId

File: app/src/main/java/com/example/spring/AddFeatureToPlanHttpRequest.java
package com.example.spring;

/**
 * Caller-visible HTTP body for command AddFeatureToPlan. Path variables and request-context fields
 * are supplied by transport state, not caller JSON.
 */
public record AddFeatureToPlanHttpRequest(
    int allowance,
    @jakarta.validation.constraints.NotBlank String effectiveAt,
    @jakarta.validation.constraints.NotBlank String enforcementMode,
    @jakarta.validation.constraints.NotBlank String entitlementId,
    @jakarta.validation.constraints.NotBlank String featureKey,
    @jakarta.validation.constraints.NotBlank String tenantId,
    @jakarta.validation.constraints.NotBlank String unit
) {}
